corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js

```diff
@@ -305,6 +305,11 @@ hqDefine("cloudcare/js/form_entry/fullform-ui", function () {
 
         self.submitForm = function () {
             $.publish('formplayer.' + Const.SUBMIT, self);
+
+            // Workaround for selectWoo. Elements are kept alive indefinitely
+            // by a keydown handler on the document, so we remove them on
+            // submission.
+            $(document).off('keydown');
```

Comment:
Would destroying the selectWoos with the select2 destroy method (assuming that works for selectWoo) achieve this? I'm a little nervous about removing all keydown handlers, as it'll prevent us from using any other code that uses a global keydown. There's one of those in our code ([here](https://github.com/dimagi/commcare-hq/blob/2fdcd034ff1e00578b0edb48a4b4dffeb0c60382/corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js#L542)) and some risk that other third-party libraries might use a document-level keydown.

It also seems like this might break selectWoos that are added after a user submits a form and then goes back to the search page. Or does the keydown event get re-added somewhere?

Reply:
...I just read the comment in the ticket where you addressed this point. Let me think about this a little bit more.

Reply:
I agree that my solution here is pretty bad and could have some unfortunate side-effects, including hard to detect bugs when some future change relies on document-level keydown events. There is probably a better way, so I'll keep trying to find one.

One option is just leaving the event handlers in place. It would be a slow memory leak but invisible to users if the selectWoos never react to the event.

A good option would be if selectWoos used a specific tag for their handlers so that only those could be removed, but I don't know how to add such a tag from outside the selectWoo code.

Regarding the question "might break selectWoos ...", there is one new handler added for every selectWoo, so every time a form is rendered and we create new selectWoos, new handlers get added. I picked submit time as a good point when existing selectWoos should no longer be usable and future ones won't have been created yet.